File: day01-数组-二分查找/35.搜索插入位置/35.搜索插入位置.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  let right = nums.length - 1;
  let left = 0;
  while (left <= right) {
    let mid = Math.floor((left + right) / 2);
    if (nums[mid] < target) {
      left = mid + 1;
    } else if (nums[mid] > target) {
      right = mid - 1;
    } else {
      return mid;
    }
    console.log(left <= right);
    return left;
  }
};

File: day01-数组-二分查找/35.搜索插入位置/35.搜索插入位置.cpp
#include <vector>;
#include <iostream>;
using namespace std;
class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int right = nums.size() - 1;
        int left = 0;
        int mid = 0;
        while (left <= right)
        {
            mid = (right + left) / 2;
            if (nums[mid] < target)
            {
                left = mid + 1;
            }
            else if (nums[mid] > target)
            {
                right = mid - 1;
            }
            else
            {
                return mid;
            }
        }
        return left;
    }
};